### PR TITLE
fix: [CI-3335]: add trigger yaml validation

### PR DESCRIPTION
**Pull request by jamie-harness**
Make sure everything in pipeline configured as run time input exist in trigger
Make sure everything in trigger exist in pipeline as run time input

You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/30351)

### CODE REVIEW

**Comment by jamie-harness** [comment]
trigger messagecheck

File: 810-ng-triggers/src/main/java/io/harness/ngtriggers/service/impl/NGTriggerServiceImpl.java [hunk 1]
```diff
@@ -562,6 +585,101 @@ public void validateTriggerConfig(TriggerDetails triggerDetails) {
...
+
+  private String getPipelineComponent(String triggerYml) {
+    try {
+      if (EmptyPredicate.isEmpty(triggerYml)) {
+        return triggerYml;
+      }
+      JsonNode node = YamlUtils.readTree(triggerYml).getNode().getCurrJsonNode();
+      ObjectNode innerMap = (ObjectNode) node.get(TRIGGER);
+      if (innerMap == null) {
+        throw new InvalidRequestException("Yaml provided is not an trigger yaml.");
```
Comment (harshjain12) [inline 810-ng-triggers/src/main/java/io/harness/ngtriggers/service/impl/NGTriggerServiceImpl.java:619]: Just Send **Invalid trigger yaml**

File: 810-ng-triggers/src/main/java/io/harness/ngtriggers/service/impl/NGTriggerServiceImpl.java [hunk 2]
```diff
@@ -562,6 +585,101 @@ public void validateTriggerConfig(TriggerDetails triggerDetails) {
...
+      }
+      JsonNode inputYaml = innerMap.get(INPUT_YAML);
+      if (inputYaml == null) {
+        throw new InvalidRequestException("Yaml provided is not an trigger yaml.");
```
Comment (harshjain12) [inline 810-ng-triggers/src/main/java/io/harness/ngtriggers/service/impl/NGTriggerServiceImpl.java:623]: Just Send Invalid trigger yaml

**Review by harshjain12 — APPROVED** [review]
:lgtm:

> Reviewed 3 of 6 files at r1, 3 of 3 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30351#-:-MuyWlcL7RU0LhYyqhWu:bnfp4nl)* status: :shipit: complete! ✓

**Review by mtran7 — APPROVED** [review]
Approving on behalf of Harsh

**Review by dgarg-harness — APPROVED** [review]

**Review by dgarg-harness — COMMENTED** [review]
> Reviewed 3 of 6 files at r1, 2 of 3 files at r2, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30351)* status: :shipit: complete! ✓

**Pull request merged**